codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -20,23 +20,152 @@
 import io.netty.handler.codec.compression.ZlibWrapper;
 import io.netty.util.internal.StringUtil;
 
+import java.util.regex.Pattern;
+
 /**
  * Compresses an {@link HttpMessage} and an {@link HttpContent} in {@code gzip} or
- * {@code deflate} encoding while respecting the {@code "Accept-Encoding"} header.
- * If there is no matching encoding, no compression is done.  For more
- * information on how this handler modifies the message, please refer to
- * {@link HttpContentEncoder}.
+ * {@code deflate} encoding while respecting the {@code "Accept-Encoding"} header. If there is no matching encoding,
+ * no compression is done.
+ * <p/>
+ * Note that only a {@link FullHttpResponse} or a response with {@code "Transfer-Encoding: chunked"}
+ * will be compressed. Also the {@code "Content-Length"} and {@code "Content-Type"} are checked against
+ * the given <tt>minCompressableContentLength</tt> and <tt>compressableContentTypes</tt> to determined if
+ * a compression would be effective at all.
+ * <p/>
+ * Note that if one of the deprecated constructors is invoked, all requests independent of their request type,
+ * transfer encoding, content type and content length. This is for backward compatibility reasons.
+ * <p/>
+ * For more information on how this handler modifies the message, please refer to {@link HttpContentEncoder}.
  */
 public class HttpContentCompressor extends HttpContentEncoder {
 
-    private final int compressionLevel;
-    private final int windowBits;
-    private final int memLevel;
+    private int compressionLevel;
+    private int windowBits;
+    private int memLevel;
+    private int minCompressableContentLength;
+    private Pattern compressableContentTypes;
+
+    /**
+     * Creates a new content compressor with default settings.
+     * <p/>
+     * Use the <tt>with...</tt> methods to further customize the settings.
+     * <p/>
+     * A compressor created by this method will only compress FullHttpResponses or responses with {@code
+     * "Transfer-Encoding: chunked"}.
+     *
+     * @return a new content compressor with sane default settings
+     */
+    public static HttpContentCompressor create() {
```

Comment:
The default constructor already exists and I don't want to change its behaviour so that it remains backward compatible.